src/Mod/Part/App/GeometryMigrationExtension.h

```diff
@@ -76,7 +102,21 @@ class PartExport GeometryMigrationExtension : public Part::GeometryExtension
     using MigrationTypeFlagType = std::bitset<32>;
     MigrationTypeFlagType           GeometryMigrationFlags;
     bool                            ConstructionState{false};
+    long                            Id = 0;
+    int                             RefIndex = -1;
+    unsigned long                   Flags = 0;
+    std::string                     Ref;
+};
+
 
+class PartExport GeometryMigrationPersistenceExtension : public Part::GeometryPersistenceExtension
+{
+    TYPESYSTEM_HEADER();
+public:
+    // Called to save data as attributes in 'Geometry' XML tag
+    virtual void preSave(Base::Writer &/*writer*/) const {}
+    // Called to save data after 'GeometryExtensions' XML elements
+    virtual void postSave(Base::Writer &/*writer*/) const {}

```

Comment:
These method names sound like things that are called before and after the "real" save (a common idiom in our codebase), but it sounds from their descriptions like that is *not* what they do. Can you either clarify the comment, or the method name?

Reply:
Adjustment made in #15497